### PR TITLE
Add support for isMetadataFilled for aggregation process

### PR DESCRIPTION
**Pull request by zyiou**
If correlation is required, we will set `isMetadataFilled` to true.
if correlation is not required, we will set `isMetadataFilled` to true only when the flow is intra-Node.

`isMetadataFilled` is always true for Intra-Node and ToExternal flows and only applicable for Inter-Node flows that are not required to be correlated. (e.g. flows with Egress deny rule applied)

### CODE REVIEW

File: pkg/intermediate/aggregate.go [hunk 1]
```diff
@@ -331,6 +340,12 @@ func (a *AggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record ent
 		}
 		if !correlationRequired {
 			aggregationRecord.ReadyToSend = true
+			// if no correlation is required for inter-Node record, K8s metadata will not be filled.
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:343]: Could you rephrase this? Also please explicitly mention that for Intra-node flows and ToExternal flows, we can set the MetaDataFilled as true by default.

**Review by srikartati — COMMENTED** [review]
Please also add explicitly that the new field is always true for Intra-Node flows and ToExternal flows in the PR description. It is only applicable for inter-node flows and we do not expect them to be correlated.

File: pkg/intermediate/aggregate.go [hunk 2]
```diff
@@ -331,6 +340,14 @@ func (a *AggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record ent
 		}
 		if !correlationRequired {
 			aggregationRecord.ReadyToSend = true
+			// if no correlation is required for inter-Node record, K8s metadata will not
+			// be filled. For Intra-Node flows and ToExternal flows, isMetaDataFilled is set
+			// to true by default.
+			if isRecordFromSrc(record) || isRecordFromDst(record) {
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:346]: Not your change? Do we still need `isRecordFromSrc` and `isRecordFromDst`? We already have flow type in the flow record. Can we use that to make code simple?

File: pkg/intermediate/aggregate.go [hunk 3]
```diff
@@ -331,6 +340,14 @@ func (a *AggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record ent
 		}
 		if !correlationRequired {
 			aggregationRecord.ReadyToSend = true
+			// if no correlation is required for inter-Node record, K8s metadata will not
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:343]: "If no correlation is required for an inter-Node record, K8s metadata is expected to be not completely filled."

File: pkg/intermediate/aggregate.go (hunk 2 above)
Comment (zyiou) [reply pkg/intermediate/aggregate.go]: Sure. We can switch to using flowType here. But function `isRecordFromSrc`, `isRecordFromDst` and `areRecordsFromSameNode` need to be kept because this information is not revealed from flowType.

Comment (srikartati) [reply pkg/intermediate/aggregate.go]: Yes we still need those methods. Moving to flowType here makes the code easily readable.

**Review by srikartati — APPROVED** [review]
Thanks for addressing comments. 
LGTM
